modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormGuestUploadFieldUtil.java

```diff
@@ -55,19 +55,23 @@ public static boolean isMaximumSubmissionLimitReached(
 				WorkflowConstants.STATUS_ANY, QueryUtil.ALL_POS,
 				QueryUtil.ALL_POS, null);
 
-		for (int i = 0; i < ddmFormInstanceRecords.size(); i++) {
-			DDMFormInstanceRecord ddmFormInstanceRecord =
-				ddmFormInstanceRecords.get(i);
+		int count = 0;
+
+		for (DDMFormInstanceRecord ddmFormInstanceRecord :
+				ddmFormInstanceRecords) {
 
 			if (Objects.equals(
 					ddmFormInstanceRecord.getIpAddress(),
-					httpServletRequest.getRemoteAddr()) &&
-				((i + 1) == guestUploadMaximumSubmissions)) {
+					httpServletRequest.getRemoteAddr())) {
 
-				return true;
+				count++;
 			}
 		}
 
+		if (count >= guestUploadMaximumSubmissions) {
+			return true;
+		}
+
 		return false;
```

Comment:
@rijema please try to simplify the code to something like:

```
int count = ListUtil.count(
	ddmFormInstanceRecordLocalService.getFormInstanceRecords(
		ddmFormInstance.getFormInstanceId(),
		WorkflowConstants.STATUS_ANY, QueryUtil.ALL_POS,
		QueryUtil.ALL_POS, null),
	ddmFormInstanceRecord -> Objects.equals(
		ddmFormInstanceRecord.getIpAddress(),
		httpServletRequest.getRemoteAddr()));

if (count >= guestUploadMaximumSubmissions) {
	return true;
}

return false;
```

Reply:
Hi @carolmariaabb , I did a small changes over this suggestion as discussed earlier.

Please, imagine the case where the customer has 1000 entries. If they doesn't change the bare minimum limit to guest entries, the system will consider the default (5). Maintaining the count, we will interact with each entry, evaluate with the Object.equals condition, and after passing through the 1000 entries, we will check if the count is higher than submission limit. I maintained the for loop and I'm getting out from the loop right when it reaches the submission limit. 
 
 It's inside the other if since there is no need to evaluate the value if there is no increase on its value.